jupyterhub/handlers/base.py

```diff
@@ -654,8 +654,36 @@ def get_next_url(self, user=None, default=None):
                     next_url = url_path_join(self.hub.base_url, 'spawn')
             else:
                 next_url = url_path_join(self.hub.base_url, 'home')
+
+        next_url = self.append_query_parameters(next_url, exclude=['next'])
         return next_url
 
+    def append_query_parameters(self, url, exclude=None):
```

Comment:
Maybe the default should be to exclude `next` since it's called that way 100% of the time so far?

Reply:
Hmmm, good point. That would prevent other developers using it and accidentally including `next`, which could trigger a redirect-loop. Let me update it.